SpellingBee/create_spelling_data.py

```diff
@@ -0,0 +1,320 @@
+import torch
+import numpy as np
+import os
+import argparse
+import itertools
+from tqdm.auto import tqdm
+
+import nltk
+from nltk.stem import WordNetLemmatizer
+from nltk.corpus import wordnet
+
+lemmatizer = WordNetLemmatizer()
+
+
+def preprocess_dic(input_dic):
+    input_dic = open(input_dic)
+    input_dic = [x.split(' ')[0] for x in input_dic]
+    return input_dic
+
+
+def normalize(token_index, cur_dict, fname):
+    '''make token varations in RoBERTa's dict into the same token '''
+
+    def get_wordnet_pos(treebank_tag):
+        if treebank_tag.startswith('J'):
+            return wordnet.ADJ
+        elif treebank_tag.startswith('V'):
+            return wordnet.VERB
+        elif treebank_tag.startswith('N'):
+            return wordnet.NOUN
+        elif treebank_tag.startswith('R'):
+            return wordnet.ADV
+        else:
+            return ''
+
+    if 'roberta' in fname:
+        token = cur_dict[token_index]
+        token = token.replace('Ġ', '').lower()
+    if 'gpt2' in fname:
+        token = cur_dict[token_index]
+        token = token.replace('Ġ', '').lower()
+    if 'AraBERT' in fname:
+        token = cur_dict[token_index]
+        token = token.replace('##', '').lower()
+        return stemmer.stem(token)
+    if 'GloVe' in fname:
+        return cur_dict[token_index]
```

Comment:
doesn't this need to be else-if?
If so consider doing `else: raise ValueError(...)`